synapse/lib/types.py

```diff
@@ -541,18 +547,27 @@ class Array(Type):
     isarray = True
     ismutable = True
 
+    _opt_defs = (
+        ('type', None),       # type: ignore
+        ('uniq', True),       # type: ignore
+        ('split', None),      # type: ignore
+        ('sorted', True),     # type: ignore
+        ('typeopts', None),   # type: ignore
+    )
+
     def postTypeInit(self):
 
-        self.isuniq = self.opts.get('uniq', True)
-        self.issorted = self.opts.get('sorted', True)
-        self.splitstr = self.opts.get('split', None)
+        self.isuniq = self.opts.get('uniq')
+        self.issorted = self.opts.get('sorted')
+        self.splitstr = self.opts.get('split')
 
         typename = self.opts.get('type')
         if typename is None:
             mesg = 'Array type requires type= option.'
             raise s_exc.BadTypeDef(mesg=mesg)
```

Comment:
Is this duplicative with the base Type check now?

Reply:
I don't think so, this is checking that a `type` opt was provided because the default is `None`, the new check is just making sure no unknown opts are provided.